ext/pdo/tests/type_error_cannot_convert_to_bool.phpt

```diff
@@ -0,0 +1,38 @@
+--TEST--
+PDO Common: TypeError when binding an object as a boolean which cannot be converted
+--EXTENSIONS--
+pdo
+gmp
+--SKIPIF--
+<?php
+$dir = getenv('REDIR_TEST_DIR');
+if (false == $dir) die('skip no driver');
+require_once $dir . 'pdo_test.inc';
+PDOTest::skip();
+?>
+--FILE--
+<?php
+if (getenv('REDIR_TEST_DIR') === false) putenv('REDIR_TEST_DIR='.__DIR__ . '/../../pdo/tests/');
+require_once getenv('REDIR_TEST_DIR') . 'pdo_test.inc';
+$db = PDOTest::factory();
+
+$db->exec('CREATE TABLE test(id int NOT NULL PRIMARY KEY, val boolean)');
+$db->exec('INSERT INTO test VALUES(1, TRUE)');
+$db->exec('INSERT INTO test VALUES(2, FALSE)');
```

Comment:
I'm not sure about this test case. I think this test case should be present in every driver separately as some of them have the boolean type and some don't. Also, binding using `PDO::PARAM_BOOL` might not be properly supported for some of them. I'd also expect this test case to happen with emulated and without emulated prepares. 

The exception checking you have added looks like the wrong solution. See MySQL execution for example, which silently fails and causes a syntax error. We don't want to do that. It's just my opinion, what's yours?

Reply:
Some of the tests in ext/pdo are skipped for certain drivers (namely OCI) and I prefer doing this because if a new driver gets added/modified we'll know this case will be handled.

The exception check might be incorrect because I blindly added it everywhere where a call which *could* throw comes up, so any improvements are welcomed. :-)

Reply:
Ok, then how about instead of using `boolean` data type, use `int` instead. I think it's more widely supported. Then also you can change `TRUE/FALSE` to `1/0`.